libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/test/MockMessageBus.scala

```diff
@@ -0,0 +1,41 @@
+package com.advancedtelematic.libats.messaging.test
+
+import java.util.concurrent.ConcurrentHashMap
+
+import akka.http.scaladsl.util.FastFuture
+import com.advancedtelematic.libats.messaging.MessageBusPublisher
+import com.advancedtelematic.libats.messaging_datatype.MessageLike
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+
+class MockMessageBus extends MessageBusPublisher {
+
+  private case class ReceivedMsg(tag: ClassTag[_], payload: Any)
+
+  private val received = new ConcurrentHashMap[String, ReceivedMsg]()
+
+  def wasReceived[T](filterFn: T => Boolean)(implicit msgLike: MessageLike[T]): Option[T] = {
+    val found: T = received.searchValues(1024, (msg: ReceivedMsg) => {
+      val maybeT = if(msg.tag.equals(msgLike.tag)) Some(msg.payload.asInstanceOf[T]) else None
```

Comment:
Can we use `==` for `ClassTag`?

Reply:
I guess we could, but why would it be better?

Reply:
It's just more readable IMO... And arguably more "Scala-y"?

Reply:
:+1: